src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/CommandFactory.shared.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+	public static class CommandFactory
+	{
+		public static Command Create(Action execute, Func<bool> canExecute = null) =>
+			new Command(execute, canExecute ?? (() => true));
+
+		public static Command Create<TExecute>(Action<TExecute> execute, Func<bool> canExecute = null) =>
+			new Command(ConvertExecute(execute), ConvertCanExecute(canExecute));
+
+		public static Command Create<TExecute, TCanExecute>(Action<TExecute> execute, Func<TCanExecute, bool> canExecute) =>
+			new Command(ConvertExecute(execute), ConvertCanExecute(canExecute));
+
+		public static IAsyncCommand Create(
+			Func<Task> execute,
+			Func<bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false,
+			bool allowsMultipleExecutions = true) =>
+			new AsyncCommand(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
+
+		public static IAsyncCommand<TExecute> Create<TExecute>(
+			Func<TExecute, Task> execute,
+			Func<bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false,
+			bool allowsMultipleExecutions = true) =>
+			new AsyncCommand<TExecute>(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
+
+		public static IAsyncCommand<TExecute, TCanExecute> Create<TExecute, TCanExecute>(
+			Func<TExecute, Task> execute,
+			Func<TCanExecute, bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false,
+			bool allowsMultipleExecutions = true) =>
+			new AsyncCommand<TExecute, TCanExecute>(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
+
```

Comment:
Let's add support for `IAsyncValueCommand`

Reply:
By support for `IAsyncValueCommand` you mean creating new `CreateValue()` method that would return value commands?
From what I now, Xamarin doesn't have `ValueCommand`, so what should this method return for sync actions?

That's why I didn`t include `IAsyncValueCommand` here in the first place - I couldn't figure out how to do it. Do you have any ideas on how this could be done?

Reply:
Here is the implementation for `IAsyncValueCommand`:
```csharp
public static IAsyncValueCommand Create(
	Func<ValueTask> execute,
	Func<bool> canExecute = null,
	Action<Exception> onException = null,
	bool continueOnCapturedContext = false,
	bool allowsMultipleExecutions = true) =>
	new AsyncValueCommand(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);

public static IAsyncValueCommand<TExecute> Create<TExecute>(
	Func<TExecute, ValueTask> execute,
	Func<bool> canExecute = null,
	Action<Exception> onException = null,
	bool continueOnCapturedContext = false,
	bool allowsMultipleExecutions = true) =>
	new AsyncValueCommand<TExecute>(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);

public static IAsyncValueCommand<TExecute, TCanExecute> Create<TExecute, TCanExecute>(
	Func<TExecute, ValueTask> execute,
	Func<TCanExecute, bool> canExecute = null,
	Action<Exception> onException = null,
	bool continueOnCapturedContext = false,
	bool allowsMultipleExecutions = true) =>
	new AsyncValueCommand<TExecute, TCanExecute>(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
```

Reply:
Oh, right. It uses different task type and can be differentiated using it. Missed that, thank you. I thought the command itself is value type.

Reply:
Hm, after adding support for `AsyncValueCommand` we have ambiguity when using async lambda as first argument.

![image](https://user-images.githubusercontent.com/6609929/105823588-052e0f00-5fc6-11eb-9e69-3ced233769e0.png)

Reply:
@brminnick Can we just support `Command` and `AsyncCommand` with `CommandFactory.Create` since those two are most commonly used onse?

Reply:
`AsyncValueCommand` is almost never used, so removed it from `CommandFactory`.

Here's what I mean by "almost never":
![image](https://user-images.githubusercontent.com/6609929/105886215-c96d6680-6012-11eb-8777-11c9e83ad92f.png)
![image](https://user-images.githubusercontent.com/6609929/105886216-ca9e9380-6012-11eb-9eeb-15928f69906e.png)

User can still create it manually if they need it

Reply:
> Can we just support Command and AsyncCommand with CommandFactory.Create since those two are most commonly used onse?

No. We cannot add a support library for `IAsyncCommand` whilst ignoring `IAsyncValueCommand`.

 I will not approve this PR without support for `ValueTask` and `IAsyncValueCommand`.

Reply:
Do you have some ideas on how to remove the ambiguity with async lambdas without forcing users to write explicit cast? The only one I came up with is to create `AsyncValueCommand` from a different method `CreateValue` or something. But it that feels kinda strange.

Reply:
With all due respect, Maksym, you are author of this PR; not me. I'd like for you to make an attempt, and I'm happy to review it.

I've already [provided you code that provides a starting point for supporting `IAsyncValueCommand`](https://github.com/xamarin/XamarinCommunityToolkit/pull/797#discussion_r564183571). I recommend massaging these provided APIs until you reach a solution.

Reply:
Thank you again for providing those APIs. 

What do you think about this approach: `Create()` is used for most commonly used `ICommand` types `Command` and `IAsyncCommand` (mainly for automatic command type resolution), and `CreateXXX()` methods are used for other types like `CreateValue()` for `IAsyncValueCommand`.

Committed changes with `CreateValue` and tests for it.

Reply:
Got it! (some kind people helped) Optional parameters can act as tiebreakers.
Added overloads for `IAsyncCommand Create` methods without optional parameters so they would be preferred over `IAsyncValueCommand Create` that does have optional parameters.